doc/source/whatsnew/v1.0.1.rst

```diff
@@ -41,6 +41,7 @@ Deprecations
 
 Bug fixes
 ~~~~~~~~~
+- Bug in :meth:`core.groupby.DataFrameGroupBy.apply` where the result shape was incorrect when the output index was not identical to the input index (:issue:`31612`)
```

Comment:
can you put this under the groupby section

Reply:
can you do this

Reply:
move to 1.1

Reply:
can you do this

Reply:
done